docker/Dockerfile

```diff
@@ -0,0 +1,72 @@
+FROM debian:jessie
+
+ENV PYTHONIOENCODING UTF-8
+
+# Pypy is installed from a package manager because it takes so long to build.
+RUN apt-get update && apt-get install -y \
+    build-essential \
+    curl \
+    git \
+    libbz2-dev \
+    libcurl4-openssl-dev \
+    libmemcached-dev \
+    libncurses5-dev \
+    libreadline-dev \
+    libsqlite3-dev \
+    libssl-dev \
+    pkg-config \
+    pypy \
+    wget \
+    zlib1g-dev
+
+# Setup variables. Even though changing these may cause unnecessary invalidation of
+# unrelated elements, grouping them together makes the Dockerfile read better.
+ENV PROVISIONING /provisioning
+ENV CELERY_USER someone
```

Comment:
Let's just change this to `developer` to reduce complexity and options. Having a consistent username will also help with possible bug reports regarding permission errors. Perhaps `docker/.env` is not required as well, if you proceed with this change.

Reply:
Developer does sound much better.

The reason `.env` is useful is so `docker-compose` can be properly provisioned with volume overrides (`- ../celery:/home/$CELERY_USER/celery`) instead of copying a chosen user everywhere. However, I forgot that build-time args can be useful for reducing duplication of definitions. I changed this to be:
```
ARGS CELERY_USER
```
which will provision the user on build time and added the build-time arg to the docker-compose file. Thus, the only place `CELERY_USER=developer` appears is in the `.env` file (and this is also the single place to update).

To prevent the inevitable bug from someone running `docker build -t xxx .` for some reason, I added in this snippet as well: `RUN : "${CELERY_USER:?CELERY_USER build argument needs to be set and non-empty.}"` that ensures this build variable is set.

This keeps the options limited and working for the traditional case